React InstantSearch Hooks Native/getting-started/src/Highlight.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Fragment } from 'react';
+import { StyleSheet, Text } from 'react-native';
+import { Hit as AlgoliaHit } from '@algolia/client-search';
+import {
+  getHighlightedParts,
+  getPropertyByPath,
+} from 'instantsearch.js/es/lib/utils';
+
+type HighlightPartProps = {
+  children: React.ReactNode;
+  isHighlighted: boolean;
+};
+
+function HighlightPart({ children, isHighlighted }: HighlightPartProps) {
+  const styles = StyleSheet.create({
+    highlight: {
+      fontWeight: isHighlighted ? 'bold' : 'normal',
+      backgroundColor: isHighlighted ? '#f5df4d' : 'transparent',
+      color: isHighlighted ? '#6f6106' : 'black',
+    },
+  });
```

Comment:
a stylesheet is meant to be created only once, not in render. We can have a variant for highlighted and non-highlighted and put the ternary between the different styles key. It could be for example:

```js
const styles = StyleSheet.create({
  highlighted: {
    fontWeight: 'bold',
    backgroundColor: '#f5df4d',
    color: '#6f6106',
  },
  nonHighlighted: {
    fontWeight: 'normal',
    backgroundColor: 'transparent',
    color: 'black',
  },
});

// ...

<Text style={isHighlighted ? styles.highlighted : styles.nonHighlighted}>
  {children}
</Text>;
```

Reply:
Good suggestion, handled in [a8fc90a](https://github.com/algolia/doc-code-samples/pull/358/commits/a8fc90ab599b0a137aad2733c6b53eb7ee3b95d6).